readthedocs/builds/utils.py

```diff
@@ -79,6 +86,33 @@ def get_vcs_url(*, project, version_type, version_name):
     return project.repo.replace('git://', 'https://').replace('.git', '') + url
 
 
+def external_version_name(build_or_version):
+    """Returns a string identifying the external build/version's nature."""
+    if not build_or_version.is_external:
+        return None
+
+    project = build_or_version.project
+
+    if project.git_provider_name == GITHUB_BRAND:
+        return GITHUB_EXTERNAL_VERSION_NAME
+
+    if project.git_provider_name == GITLAB_BRAND:
+        return GITLAB_EXTERNAL_VERSION_NAME
+
+    # TODO: Add External Version Name for BitBucket.
+    return GENERIC_EXTERNAL_VERSION_NAME
+
+
+def abbrev(string):
+    """
+    Returns an abbreviation for the given string.
+
+    The abbreviation is formed for the first letter in each word in the string,
+    capitalized.
+    """
+    return ''.join(word[0].upper() for word in string.split())
```

Comment:
I think this one can also be moved to where the function is used. But I'm also not sure what's best having `PR` or `Pull Request`